client/src/components/App.js

```diff
@@ -1,25 +1,28 @@
-import data from '../data/products.json'
+//import json from '../data/products.json'
 import styled from 'styled-components/macro'
+import { useQuery } from 'react-query'
+
+import { Switch, Route } from 'react-router-dom'
+
 import ProductListing from './ProductList/ProductList'
 import ProductDetail from './ProductDetail/ProductDetail'
 
-import { Switch, Route } from 'react-router-dom'
+import { getProducts } from '../services/getProducts'
 
 export default function App() {
+  const { status, data } = useQuery('products', () => getProducts())
   return (
     <AppGrid>
+      {status === 'loading' && 'Loading...'}
       <Switch>
         <Route exact path="/">
           <ProductsContainer>
-            {data.map(product => (
-              <ProductListing key={product.id} product={product} />
+            {data?.map(product => (
+              <ProductListing key={product._id} product={product} />
             ))}
           </ProductsContainer>
         </Route>
-        <Route
-          path="/:styleId"
-          render={props => <ProductDetail {...props} />}
-        />
+        <Route path="/:styleId" children={<ProductDetail />} />
```

Comment:
Why not `productId` or just `id`? It is the id of the product, right? `styleId` sounds a bit like it has something to do with styling/css. But that's just my personal opinion.

Reply:
styleId is a fancy word for SKU. it's unique but ist not like the id

Reply:
Learned something new! Thanks :)